src/main/kotlin/com/cognifide/gradle/aem/common/file/FileOperations.kt

```diff
@@ -19,31 +19,57 @@ import org.zeroturnaround.zip.ZipUtil
 
 object FileOperations {
 
-    fun readResource(path: String): InputStream? {
+    fun <T> readResourceFromPathOrClasspath(resource: String, parser: (BufferedReader) -> T): T =
+            BufferedReader(InputStreamReader(inputStreamFromPathOrClasspath(resource))).use(parser)
+
+    fun <T> optionalReadResourceFromPathOrClasspath(resource: String, parser: (BufferedReader) -> T): T? =
+            try {
+                readResourceFromPathOrClasspath(resource, parser)
+            } catch (e: AemException) {
+                null
+            }
+
+    private fun inputStreamFromPathOrClasspath(resourcePath: String): InputStream {
+        val resourceFile = File(resourcePath)
+        if (resourceFile.exists()) {
+            return GFileUtils.openInputStream(resourceFile)
+        }
+        val resourceStream: InputStream? = this::class.java.getResourceAsStream(resourcePath)
+        if (resourceStream != null) {
+            return resourceStream
+        }
+        val resource: URL? = this::class.java.classLoader.getResource(resourcePath)
+        if (resource != null) {
+            return GFileUtils.openInputStream(File(resource.file))
+        }
+        throw AemException("Cannot load blacklist from file: $resourcePath")
+    }
+
+    fun readResourceFromAemPkg(path: String): InputStream? {
```

Comment:
`fromAemPkg` / too detailed for me. we are not doing it in a different way? don't we?

Reply:
OK